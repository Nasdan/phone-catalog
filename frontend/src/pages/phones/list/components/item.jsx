import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'emotion';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';
import { item, card, cardContent, image, title, price } from './item.styles';

export const Item = props => (
  <li className={cx(item, props.className)}>
    <Card className={card}>
      <CardContent className={cardContent}>
        <div className={image}>
          <img src={props.phone.imageUrl} />
        </div>
        <Typography className={title}>{props.phone.title}</Typography>
        <Typography className={price}>
          <h2>{props.phone.price}</h2>
        </Typography>
      </CardContent>
    </Card>
  </li>
);

Item.propTypes = {
  phone: PropTypes.object.isRequired,
  className: PropTypes.string.isRequired,
};
