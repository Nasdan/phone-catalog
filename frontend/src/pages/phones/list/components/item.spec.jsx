import React from 'react';
import CardActionArea from '@material-ui/core/CardActionArea';
import { shallow } from 'enzyme';
import { Item } from './item';

describe('pages/phones/list/components/item specs', () => {
  it('should render as expected when feeding props', () => {
    // Arrange
    const props = {
      phone: {
        title: 'test title',
        imageUrl: 'test imageUrl',
        price: 'test price',
      },
      className: 'test className',
      onClick: jest.fn(),
    };

    // Act
    const component = shallow(<Item {...props} />);

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should call to onClick prop with phone property when simulate CardActionArea click', () => {
    // Arrange
    const props = {
      phone: {
        title: 'test title',
        imageUrl: 'test imageUrl',
        price: 'test price',
      },
      className: 'test className',
      onClick: jest.fn(),
    };

    // Act
    const component = shallow(<Item {...props} />);

    const cardActionAreaComponent = component.find(CardActionArea);
    cardActionAreaComponent.simulate('click');

    // Assert
    expect(props.onClick).toHaveBeenCalled();
    expect(props.onClick).toHaveBeenCalledWith(props.phone);
  });
});
