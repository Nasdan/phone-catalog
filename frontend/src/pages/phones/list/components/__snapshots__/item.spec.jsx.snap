// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/phones/list/components/item specs should render as expected when feeding props 1`] = `
<li
  className="css-o9b79t test className"
>
  <WithStyles(Card)
    className="css-a0h9s2"
  >
    <WithStyles(CardActionArea)
      className="css-8atqhb"
      onClick={[Function]}
    >
      <WithStyles(CardContent)
        className="css-dvxtzn"
      >
        <div
          className="css-1fv1iua"
        >
          <img
            src="test imageUrl"
          />
        </div>
        <WithStyles(Typography)
          className="css-56sg73"
        >
          test title
        </WithStyles(Typography)>
        <h2
          className="css-56sg73"
        >
          test price
        </h2>
      </WithStyles(CardContent)>
    </WithStyles(CardActionArea)>
  </WithStyles(Card)>
</li>
`;
